docs/create_rules.md

```diff
@@ -91,30 +113,48 @@ def sql_has_reasonable_number_of_lines(model: Model, max_lines: int = 200) -> Ru
         )
 ```
 
-### Filtering models
+### Filtering rules
 
-Custom and standard rules can be configured to have model filters. Filters allow
-models to be ignored by one or multiple rules.
+Custom and standard rules can be configured to have filters. Filters allow
+models or sources to be ignored by one or multiple rules if the item doesn't satisfy
+the filter criteria.
 
 Filters are created using the same discovery mechanism and interface as custom
 rules, except they do not accept parameters. Similar to Python's built-in
-`filter` function, when the filter evaluation returns `True` the model should be
+`filter` function, when the filter evaluation returns `True` the item should be
 evaluated, otherwise it should be ignored.
 
 ```python
-from dbt_score import ModelFilter, model_filter
+from dbt_score import Model, RuleFilter, rule_filter
 
-@model_filter
+@rule_filter
 def only_schema_x(model: Model) -> bool:
     """Only applies a rule to schema X."""
     return model.schema.lower() == 'x'
 
-class SkipSchemaY(ModelFilter):
+class SkipSchemaY(RuleFilter):
     description = "Applies a rule to every schema but Y."
     def evaluate(self, model: Model) -> bool:
       return model.schema.lower() != 'y'
 ```
 
+Filters also rely on type-annotations to dictate whether they apply to models or sources:
```

Comment:
I'm realizing now that I haven't actually made any changes to enforce that resource_types match across `rule_filter`s attached to `rule`s.  

This would be easy to account for by addressing it in the `rule.should_evaluate` method, but would it be better to throw an error earlier with a nice message such that something like this can't be configured at all?

```python
    @rule_filter
    def source_schema_in_x(source: Source):
        """Only keep sources in schema x."""
        return schema_in_x(source)

    @rule(rule_filters={source_schema_in_x()})
    def model_has_description(model: Model):
        """Validates that a model has a description."""
        if not model.description:
            return RuleViolation("Model does not have a description.")
```

Reply:
I agree throwing an error earlier makes more sense indeed, it improves UX in my opinion! :+1:

Reply:
addressed in 0eeae68

Reply:
Great :raised_hands: